torch/_inductor/codegen/triton.py

```diff
@@ -2524,9 +2525,24 @@ def final_argreduce(buffer, result_var, value, index):
                 {result_var} = {self.reduction_resize(f"{result_var}_idx")}
                 """
             )
+        
+        #if reduction_type in ('max') and reduction_type in ('argmax'):
+        #    reduction_type = "max_argmax"
+        TritonKernel.current_reductions.append((src_dtype, reduction_type, value))
 
-        cache_key = (src_dtype, reduction_type, value)
+
+        reduction_type_search = reduction_type
+        reduction_analog_map = {
+            "argmax":"max"
```

Comment:
Won't this cause the argmax not to be computed?  I think you need to:
1) Map it the other way around (argmax computes both max+argmax, while max only computes max)
2) If doing a max, select the (currently unused output) from the argmax call, which involves changing the codegen
3) Add some tests to ensure we are generating the combined max+argmax code
4) [Optimization] In the case the only the max is used, don't (wastefully) also compute the argmax